src/prefect/client/schemas/objects.py

```diff
@@ -212,40 +211,29 @@ class State(ObjectBaseModel, Generic[R]):
     ] = Field(default=None)
 
     @overload
-    def result(
+    async def result(
         self: "State[R]",
         raise_on_failure: Literal[True] = ...,
-        fetch: bool = ...,
         retry_result_failure: bool = ...,
     ) -> R: ...
 
     @overload
-    def result(
+    async def result(
         self: "State[R]",
         raise_on_failure: Literal[False] = False,
-        fetch: bool = ...,
         retry_result_failure: bool = ...,
     ) -> Union[R, Exception]: ...
 
     @overload
-    def result(
+    async def result(
         self: "State[R]",
         raise_on_failure: bool = ...,
-        fetch: bool = ...,
         retry_result_failure: bool = ...,
     ) -> Union[R, Exception]: ...
 
-    @deprecated.deprecated_parameter(
-        "fetch",
-        when=lambda fetch: fetch is not True,
-        start_date="Oct 2024",
-        end_date="Jan 2025",
-        help="Please ensure you are awaiting the call to `result()` when calling in an async context.",
-    )
-    def result(
+    async def result(
```

Comment:
changing this to async seems like a big change?

Reply:
It was partially async before (passing fetch=True would make it async) and it would complain if it was called from an async context incorrectly. I can add an `async_dispatch` here to maintain the current behavior while removing the `fetch` kwarg.